File: pinyougou-sellergoods-interface/src/main/java/com/pinyougou/sellergoods/service/BrandService.java
package com.pinyougou.sellergoods.service;

import java.util.List;
import java.util.Map;

import com.pinyougou.pojo.TbBrand;

import entity.PageResult;

/**
 * 品牌列表
 * 
 * @author Administrator
 *
 */
public interface BrandService {
	public List<TbBrand> getAll();

	/**
	 * 品牌分页
	 * 
	 * @param pageNum  当前页面
	 * @param pageSize 当前页面记录数
	 * @return
	 */
	public PageResult findPage(int pageNum, int pageSize);

	/**
	 * 增加品牌
	 * 
	 * @param brand
	 */
	public void add(TbBrand brand);
	
	/**
	 * 修改
	 */
	public void update(TbBrand brand);
	/**
	 * 根据ID获取实体
	 * @param id
	 * @return
	 */
	public TbBrand findOne(Long id);
	
	/**
	 * 批量删除
	 * @param ids
	 */
	public void delete(Long[] ids);
	
	/**
	 * 分页
	 * @param pageNum 当前页 码
	 * @param pageSize 每页记录数
	 * @return
	 */
	public PageResult findPage(TbBrand brand, int pageNum,int pageSize);

	List<Map> selectOptionList();

}

File: pinyougou-manager-web/src/main/webapp/js/service/brandService.js
app.service('brandService', function($http) {
		//读取列表数据绑定到表单中
		this.findAll = function() {
			return $http.get('../brand/findAll.do');
		}
		this.findPage = function(page, rows) {
			return $http.get('../brand/findPage.do?page=' + page + '&rows='
					+ rows);
		}
		this.findOne = function(id) {
			return $http.get('../brand/findOne.do?id=' + id);
		}
		this.add = function(entity) {
			return $http.post('../brand/add.do', entity);
		}
		this.update = function(entity) {
			return $http.post('../brand/update.do', entity);
		}
		this.dele = function(selectIds) {
			return $http.get('../brand/delete.do?ids=' + selectIds);
		}
		this.search = function(page, rows, searchEntity) {
			return $http.post('../brand/search.do?page=' + page + "&rows="
					+ rows, searchEntity);
		}
		//下拉列表数据
		this.selectOptionList=function(){
			return $http.get('../brand/selectOptionList.do');
		}

	});